File: Medium/Q0165 - Compare Version Numbers/cpp/q0165.cpp
#include <string>
#include <cstdio>
#include <vector>

using namespace std;

int compareVersion(string version1, string version2) {
    int i = 0, j = 0;
    while (i<version1.length() || j<version2.length()) {
        int val1 = 0, val2 = 0;
        while (i<version1.length() && version1[i]!='.') {
            val1 = val1 * 10 + version1[i] - '0';
            i++;
        }
        while (j<version2.length() && version2[j]!='.') {
            val2 = val2 * 10 + version2[j] - '0';
            j++;
        }
        if (val1 > val2) {
            return 1;
        }
        else if (val1 < val2) {
            return -1;
        }
        i++;
        j++;
    }
    return 0;
}


int main() {
    vector<string> testcase_ver1 = {"1.0", "1.01", "0.1"};
    vector<string> testcase_ver2 = {"1.0.0", "1.001", "1.1"};
    for (int i=0; i< testcase_ver1.size(); i++) {
        printf("%s vs. %s? => %d\n", testcase_ver1[i].c_str(), testcase_ver2[i].c_str(), compareVersion(testcase_ver1[i], testcase_ver2[i]));
    }
    return 0;
}

File: Medium/Q0165 - Compare Version Numbers/python/q0165.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        version1 = version1.split('.')
        version2 = version2.split('.')
        i = j = 0
        while i < len(version1) or j < len(version2):
            val1 = 0 if i >= len(version1) else int(version1[i])
            val2 = 0 if j >= len(version2) else int(version2[j])
            if val1 > val2:
                return 1
            elif val1 < val2:
                return -1
            i += 1
            j += 1
        return 0